pyuvdata/tests/__init__.py

```diff
@@ -5,19 +5,245 @@
 """Setup testing environment, define useful testing functions.
 
 """
-import warnings
+import re
 import sys
+import warnings
 
 from astropy.utils import iers
 
 import pyuvdata.utils as uvutils
 
 __all__ = [
+    "check_warnings",
     "clearWarnings",
     "checkWarnings",
 ]
 
 
+class WarningsChecker(warnings.catch_warnings):
+    """
+    A context manager to check raised warnings.
+
+    Adapted from pytest WarningsRecorder and WarningsChecker.
+    """
+
+    def __init__(self, expected_warning, match):
+        super().__init__(record=True)
+        self._entered = False
+        self._list = []
+
+        assert isinstance(expected_warning, list)
```

Comment:
This assert statement appears to be in direct conflict with the `if/else` statements that follows. One of the cases is a list instance but the others could never trigger. Or am I reading this incorrectly?

Reply:
hmm, you're right. I think this is debugging cruft.